experiments/download_ssl4eo.py

```diff
@@ -647,7 +641,7 @@ def worker(idx: int) -> None:
             if ext_flags[str(idx)] != 0:  # only skip downloaded ids
                 return
 
-        patches_s1, patches_s2c, patches_s2a, center_coord = get_random_patches_grid(
+        (patches_s1, patches_s2c, patches_s2a, center_coord) = get_random_patches_rtree(
```

Comment:
It seems like we're still calling this in parallel for each worker. Sampling should occur in serial, preferably in a separate script.

Reply:
yes that would be the next pull request

Reply:
It might be useful to do that PR first (see https://github.com/microsoft/torchgeo/pull/1209#pullrequestreview-1367415789). Right now, the logic is very hard to follow.

Reply:
alright, I have that already. One thing is that includes extending sentinel-1/2 to other general products, no longer separate PR though. Is it OK like that?

Reply:
It would be better to keep each PR small and dedicated to a single change. Otherwise it becomes very difficult to review.

Reply:
ok i'll do that